modules/farm_fd2_school/src/entrypoints/FD2_Application/App.vue

```diff
@@ -182,24 +182,46 @@ export default {
         this.unit.attributes.name
       );
       console.log(quantity);
+
+      const plantAsset = await farmosUtil.getPlantAsset(this.pickedPlant.uuid);
+      console.log('Plant asset:', plantAsset);
+
+      const locationName = this.pickedPlant.location || '';
+      const bedNames = Array.isArray(this.pickedPlant.beds)
+        ? this.pickedPlant.beds
+        : this.pickedPlant.beds
+        ? [this.pickedPlant.beds]
+        : [];
+
+      const harvestLog = await farmosUtil.createdHarvestLog(
+        this.date,
+        locationName,
+        bedNames,
+        plantAsset,
+        quantity,
+        this.comment
+      );
+      console.log(harvestLog);
     },
   },
   watch: {
-    async crop() {
-      if (this.crop) {
-        const URL =
-          'http://farmos/api/fd2_plant_assets?crop=' +
-          this.crop.attributes.name;
-        const plantsResponse = await fetch(URL);
-        const plants = await plantsResponse.json();
-        if (Array.isArray(plants)) {
-          this.plantList = plants;
-        } else {
-          this.plantList = [];
-        }
-      } else {
-        this.plantList = [];
-      }
+    async crop(newCrop) {
+      this.plantList = [];
+      if (!newCrop) return;
+
+      const plants = await farmosUtil.getPlantAssets(
+        '',
+        [],
+        newCrop.attributes.name
+      );
+      this.plantList = Array.isArray(plants) ? plants : [];
+
+      const units = await farmosUtil.getHarvestUnits(newCrop.attributes.name);
+      TouchList.unitList = (units || []).map((u, i) => {
```

Comment:
TouchList has to do with a touch surface user interface? What are you trying to do with it here?